mmv1/third_party/terraform/services/compute/resource_compute_instance.go.erb

```diff
@@ -1544,6 +1543,24 @@ func resourceComputeInstanceUpdate(d *schema.ResourceData, meta interface{}) err
 	// Enable partial mode for the resource since it is possible
 	d.Partial(true)
 
+	if d.HasChange("name") {
```

Comment:
IIUC we haven't stopped the instance yet at this point- I think we need to do this below `needToStopInstanceBeforeUpdating`

Reply:
yes, moving everything down there...